PsfShimMonitor/KernelTrace.cs

```diff
@@ -0,0 +1,1356 @@
+﻿//-------------------------------------------------------------------------------------------------------
+// Copyright (C) TMurgent Technologies. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//-------------------------------------------------------------------------------------------------------
+//
+// NOTE: PsfMonitor is a "procmon"-like display of events captured via the PSF TraceShim.
+
+using Microsoft.Diagnostics.Tracing;  // consumer
+using Microsoft.Diagnostics.Tracing.Parsers;
+using Microsoft.Diagnostics.Tracing.Session; // controller
+using System;
+using System.ComponentModel;  // backgroundworker
+using System.Threading;
+using System.Windows;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;  // ObservableCollection
+
+namespace PsfMonitor
+{
+    public partial class MainWindow : Window
+    {
+        public TraceEventSession myTraceEventSession_ProcsKernel = null;
+        public bool rememberToDisableSession_ProcsKernel = false;
+        public bool PleaseStopCollecting = false;
+        public int FilterOnProcessId = -1;
+
+        public List<EventItem> _TKernelEventListItems = new List<EventItem>();
+        public Object _TKernelEventListsLock = new Object();
+
+        Dictionary<UInt64, string> _KCBs = new Dictionary<UInt64, string>();
```

Comment:
What does the CB in KCB stand for?  Can you please expand this acronym?

Reply:
Done.  Hope you like the extra long function name it caused.